modules/calib3d/test/test_cameracalibration.cpp

```diff
@@ -2158,14 +2162,23 @@ TEST(CV_RecoverPoseTest, regression_15341)
 
             // Estimation of fundamental matrix using the RANSAC algorithm
             Mat E, E2, R, t;
+
+            // Check pose when camera matrices are different.
+            E = findEssentialMat(points1, points2, cameraMatrix, zeroDistCoeffs, cameraMatrix2, zeroDistCoeffs, RANSAC, 0.999, 1.0, mask);
+            Inliers = recoverPose(points1, points2, cameraMatrix, zeroDistCoeffs, cameraMatrix2, zeroDistCoeffs, E2, R, t, RANSAC, 0.999, 1.0, mask);
```

Comment:
> zeroDistCoeffs

Non-zero distortion case should be tested too as new API suggests these parameters.